common.go

```diff
@@ -0,0 +1,74 @@
+package ibk
+
+import (
+	"bytes"
+	"context"
+	"math/rand"
+	"strings"
+	"sync"
+	"time"
+)
+
+// CombinedWriter is a writer that writes to a buffer and is safe for concurrent use.
+type CombinedWriter struct {
```

Comment:
quick-drive-by/super-nitpick: I personally would include "buffer" instead of "writer" in the name (ie. its not a wrapper for a general writer that then becomes concurrency safe). It is used as a combined writer but `common.CombinedWriter` feels to me like its not fully capturing what it is, maybe `common.SyncedBuffer`, `ThreadSafeBuffer`, `ConcurrentBuffer`, `CBuffer` or similar.

Reply:
That is a good point, added a commit.